DataPlotly/core/plot_factory.py

```diff
@@ -223,7 +223,7 @@ def add_source_field_or_expression(field_or_expression):
         stroke_colors = []
         stroke_widths = []
         for f in it:
-            if visible_geom_engine and not visible_geom_engine.intersects(f.geometry().constGet()):
+            if visible_geom_engine and not visible_geom_engine.within(f.geometry().constGet()):
```

Comment:
I'd caution against this change -- elsewhere we use intersects for testing comparison to the atlas feature (e.g. attribute table rows, legend items), so I'd be concerned this will cause inconsistencies between different layout elements.